test/micro/org/openjdk/bench/vm/compiler/CodeCacheStress.java

```diff
@@ -0,0 +1,445 @@
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+package org.openjdk.bench.vm.compiler;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Method;
+import java.util.*;
+import java.util.concurrent.ThreadLocalRandom;
+
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.CompilerControl;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Param;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.Setup;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.Threads;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.infra.Blackhole;
+
+import org.openjdk.bench.util.InMemoryJavaCompiler;
+
+@State(Scope.Benchmark)
+@Warmup(iterations = 20, time = 2)
+@Measurement(iterations = 15, time = 2)
+@BenchmarkMode(Mode.AverageTime)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@Threads(1)
+@Fork(value = 2)
+public class CodeCacheStress {
+
+    // The number of distinct classes generated from the source string below
+    // All the classes are "warmed up" by invoking their methods to get compiled by the jit
+    @Param({"100"})
+    public int numberOfClasses;
+
+    // The range of these classes to use in the measured phase after the warm up
+    @Param({"100"})
+    public int rangeOfClasses;
+
+    // How deep is the recursion when calling into the generated classes
+    @Param({"20"})
+    public int recurse;
+
+    // How many instances of each generated class to create and call in the measurement phase
+    @Param({"100"})
+    public int instanceCount;
+
+    byte[][] compiledClasses;
+    Class[] loadedClasses;
+    String[] classNames;
+
+    int index = 0;
```

Comment:
Why this `index` is here? It seems not be used except that within the `setupClasses`.

Reply:
Using index in this way to coordinate creating classes named like "B"+index is a convention that we have used in earlier tests such as hotspot/jtreg/serviceability/jvmti/RedefineClasses/TestMultipleClasses.java